src/Microsoft.OpenApi/Extensions/EnumExtensions.cs

```diff
@@ -42,5 +43,88 @@ public static string GetDisplayName(this Enum enumValue)
             var attribute = enumValue.GetAttributeOfType<DisplayAttribute>();
             return attribute == null ? enumValue.ToString() : attribute.Name;
         }
+
+        /// <summary>
+        /// Gets the enum display for name <see cref="ParameterStyle" /> without the use of reflection.
+        /// </summary>
+        /// <param name="parameterStyle">The enum value.</param>
+        /// <returns>The display string to use.</returns>
+        internal static string GetDisplayName(this ParameterStyle parameterStyle) => parameterStyle switch
+            {
+                ParameterStyle.Matrix => "matrix",
+                ParameterStyle.Label => "label",
+                ParameterStyle.Form => "form",
+                ParameterStyle.Simple => "simple",
+                ParameterStyle.SpaceDelimited => "spaceDelimited",
+                ParameterStyle.PipeDelimited => "pipeDelimited",
+                ParameterStyle.DeepObject => "deepObject",
+                _ => parameterStyle.ToString()
```

Comment:
I'd argue we should not have default cases. This way anybody adding new members will get a compile time error to come fix things here.

Reply:
I can replace these default cases with `IOE` exceptions instead.

Reply:
That'd be trading a compile time failure for a runtime one. I'm not sure it'd be better?

Reply:
Hmmm....I'm not sure I understand what you are proposing? We need a default case here in order to comply with the pattern exhaustiveness checks on the switch expression. What kind of delta were you hoping to see here?

Reply:
No worries -- your comment did help me see that I was missing a handle for `ReferenceType.Path` here. I added it and also added a set of unit tests to verify our exhaustiveness [here](https://github.com/microsoft/OpenAPI.NET/pull/1717/commits/8ea0964d6f3e1ab6ea70cb1b7e925b227513288a).

Reply:
yes! I really like this unit tests approach. At least we don't run the risk of adding a new enum value and forgetting to update the serialization code. Resolving.